### DISCUSSION TITLE
Object annotation, adjacency list querying, filter with an array

### DISCUSSION THREAD

**Discussion by tasops** [prisma/prisma#4520] [Q&A] [answered]
Howdy. I have three problems with Prisma, that i think when made into a feature, would be a crucial addition to this great project.

1. I can't find anything about **annotating** custom data/aggregations to objects when querying them. [A great example can be found in Django.](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#annotate) That's one thing i really miss coming from that framework.

2. Querying **adjacency lists** (getting descendants/ancestors etc.). For now i have no other option than to use $queryRaw property to write SQL CTE queries. It would be great if there was a functionality to do just that.

3. And last, there is no option to **filter with an array**. I also "fixed" that with just writing raw SQL. Example: `SELECT * FROM table WHERE id IN (1, 2, 3)`

**Comment by ryands17** [comment, marked as answer]
@Telluu 👋 

For the 1st point, I would suggest creating a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with your use case and the above example as that's currently not possible with Prisma.

For the 2nd point, if you're looking for descendants just with a fixed amount of levels, you can just keep on nesting with [include](https://www.prisma.io/docs/concepts/components/prisma-client/field-selection#include) to get all the levels. For infinite levels, you can use a recursive CTE and that's not possible with Prisma currently so again it would be great to create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for the same.

The last point i.e. filtering with an array is completely possible right now with Prisma.

For e.g. I have a model like this:

```prisma
model User {
  id Int @id @default(autoincrement())
}
```

I can write:

```ts
  await prisma.user.findMany({ where: { id: { in: [1, 2, 3] } } })
```
The same as your SQL example.

**Reply by tasops** [reply]
Should i create separate requests for each feature or just lump them together to one feature request? Asking because im still a noob dev and never did anything like this before.

**Reply by ryands17** [reply]
Separate requests for each feature would be best :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
